### PR TITLE
BibFormat: update KEK scan URL

### PR DESCRIPTION
**Pull request by kaplun**
* Updates KEK scan URL algorithm to reflect the recent change.
  (See https://rt.inspirehep.net/Ticket/Display.html?id=607395)

* Amends INIS algorithm to not do useless replaces.

Signed-off-by: Samuele Kaplun <samuele.kaplun@cern.ch>

### CODE REVIEW

**Comment by tsgit** [comment]
LGTM

:+1:

File: bibformat/format_elements/bfe_INSPIRE_links.py [hunk 1]
```diff
@@ -71,8 +71,14 @@ def format_element(bfo, default='', separator='; ', style='',
...
         if provenance == 'KEKSCAN':
             extid = extid.replace("-", "")
-            links.append('<a%s href="%s%s"> KEK scanned document</a>' %
-                         (style, KEK, extid))
+            if not extid.startswith('20'):
+                year = '19' + extid[:2]
+            else:
+                year = extid[:4]
+                extid = extid[2:]
+            prefix = exitd[:4]
```
Comment (tsgit) [inline bibformat/format_elements/bfe_INSPIRE_links.py]: typo  exitd => extid

**Comment by tsgit** [comment]
tested:
```
from invenio.search_engine import perform_request_search
from invenio.bibformat_engine import BibFormatObject
from invenio.bibformat_elements import bfe_INSPIRE_links
kek = perform_request_search(p="035:KEKSCAN")
len(kek)
from random import sample
for r in sample(kek, 1000):
    print bfe_INSPIRE_links.format_element(BibFormatObject(r))
```

File: bibformat/format_elements/bfe_INSPIRE_links.py (hunk 1 above)
Comment (tsgit) [inline bibformat/format_elements/bfe_INSPIRE_links.py]: another snafu: "prefix" has some other meaning in bibformat and leads to a prefix being printed in front of the format element. Doh!

**Comment by tsgit** [comment]
after fixing the 2 code issues,  I get many links that don't resolve, e.g.
https://lib-extopc.kek.jp/preprints/PDF/1919/1990/199007322.pdf
 ??


```
$ ipython
Python 2.7.12 (default, Sep 29 2016, 13:30:34) 
Type "copyright", "credits" or "license" for more information.

IPython 5.1.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: from invenio.search_engine import perform_request_search

In [2]: from invenio.bibformat_engine import BibFormatObject

In [3]: from invenio.bibformat_elements import bfe_INSPIRE_links

In [4]: from random import sample

In [6]: kek = perform_request_search(p="035:KEKSCAN")

In [7]: for r in sample(kek, 100):
   ...:     print r, bfe_INSPIRE_links.format_element(BibFormatObject(r))
   ...:     print "\n"
   ...:     
253504 <a href="https://lib-extopc.kek.jp/preprints/PDF/1988/8804/8804284.pdf"> KEK scanned document</a>; <a  href="http://alice.cern.ch/format/showfull?sysnb=0097982">CERN Library Record</a>


288818 <a href="https://lib-extopc.kek.jp/preprints/PDF/1919/1990/199007322.pdf"> KEK scanned document</a>


301584 <a href="https://lib-extopc.kek.jp/preprints/PDF/1919/1990/199012240.pdf"> KEK scanned document</a>; <a  href="http://alice.cern.ch/format/showfull?sysnb=0127718">CERN Library Record</a>


203712 <a href="https://lib-extopc.kek.jp/preprints/PDF/1984/8409/8409096.pdf"> KEK scanned document</a>; <a href="http://cds.cern.ch/record/154029"> CERN Document Server</a>; <a  href="http://alice.cern.ch/format/showfull?sysnb=0065285">CERN Library Record</a>


91148 <a href="https://lib-extopc.kek.jp/preprints/PDF/1919/1975/197503094.pdf"> KEK scanned document</a>; <a href="http://adsabs.harvard.edu/abs/1975ApL....16...57B"> ADS Abstract Service</a>


164729 <a href="https://lib-extopc.kek.jp/preprints/PDF/1981/8105/8105176.pdf"> KEK scanned document</a>


320369 <a href="https://lib-extopc.kek.jp/preprints/PDF/1991/9111/9111463.pdf"> KEK scanned document</a>; <a href="http://adsabs.harvard.edu/abs/1992PhRvL..68.2433A"> ADS Abstract Service</a>; <a  href="http://lss.fnal.gov/cgi-bin/find_paper.pl?pub-91-267">Fermilab Library Server (fulltext available)</a>; <a  href="http://inspirehep.net/record/320369/files/pub-91-267.pdf">Link to Fulltext</a>


340964 <a href="https://lib-extopc.kek.jp/preprints/PDF/1992/9212/9212388.pdf"> KEK scanned document</a>; <a href="http://adsabs.harvard.edu/abs/1993PhRvL..70.1912M"> ADS Abstract Service</a>


450192 <a href="https://lib-extopc.kek.jp/preprints/PDF/1997/9710/9710153.pdf"> KEK scanned document</a>; <a  href="http://alice.cern.ch/format/showfull?sysnb=0272861">CERN Library Record</a>


266813 <a href="https://lib-extopc.kek.jp/preprints/PDF/1988/8810/8810231.pdf"> KEK scanned document</a>; <a href="http://adsabs.harvard.edu/abs/1989PThPh..81..266K"> ADS Abstract Service</a>


457108 <a href="https://lib-extopc.kek.jp/preprints/PDF/1999/9905/9905051.pdf"> KEK scanned document</a>


382427 <a href="https://lib-extopc.kek.jp/preprints/PDF/1995/9502/9502265.pdf"> KEK scanned document</a>; <a  href="http://alice.cern.ch/format/showfull?sysnb=0196438">CERN Library Record</a>


21380 <a href="https://lib-extopc.kek.jp/preprints/PDF/1987/8704/8704354.pdf"> KEK scanned document</a>; <a href="http://adsabs.harvard.edu/abs/1987PhLB..194..530N"> ADS Abstract Service</a>


214017 <a href="https://lib-extopc.kek.jp/preprints/PDF/1985/8506/8506023.pdf"> KEK scanned document</a>; <a href="http://adsabs.harvard.edu/abs/1985PhRvL..55.2362O"> ADS Abstract Service</a>; <a  href="http://lss.fnal.gov/cgi-bin/find_paper.pl?pub-85-043-A">Fermilab Library Server (fulltext available)</a>; <a  href="http://inspirehep.net/record/214017/files/pub-85-043-A.pdf">Link to Fulltext</a>


262692 <a href="https://lib-extopc.kek.jp/preprints/PDF/1988/8807/8807160.pdf"> KEK scanned document</a>


30429 <a href="https://lib-extopc.kek.jp/preprints/PDF/1991/9112/9112123.pdf"> KEK scanned document</a>; <a href="http://adsabs.harvard.edu/abs/1991PhLB..273..316B"> ADS Abstract Service</a>


144047 <a href="https://lib-extopc.kek.jp/preprints/PDF/1980/8001/8001345.pdf"> KEK scanned document</a>; <a href="http://cds.cern.ch/record/133772"> CERN Document Server</a>


17983 <a href="https://lib-extopc.kek.jp/preprints/PDF/1986/8605/8605220.pdf"> KEK scanned document</a>; <a  href="http://alice.cern.ch/format/showfull?sysnb=0076916">CERN Library Record</a>


219525 <a href="https://lib-extopc.kek.jp/preprints/PDF/1986/8602/8602034.pdf"> KEK scanned document</a>; <a href="http://adsabs.harvard.edu/abs/1986PThPh..75..934I"> ADS Abstract Service</a>


239345 <a href="https://lib-extopc.kek.jp/preprints/PDF/2000/0031/0031919.pdf"> KEK scanned document</a>


4966 <a href="https://lib-extopc.kek.jp/preprints/PDF/1919/1977/197702178.pdf"> KEK scanned document</a>


317297 <a href="https://lib-extopc.kek.jp/preprints/PDF/1991/9108/9108172.pdf"> KEK scanned document</a>; <a  href="http://alice.cern.ch/format/showfull?sysnb=0136617">CERN Library Record</a>


426726 <a href="https://lib-extopc.kek.jp/preprints/PDF/1996/9612/9612062.pdf"> KEK scanned document</a>; <a href="http://adsabs.harvard.edu/abs/1998PPN....29..463G"> ADS Abstract Service</a>; <a  href="http://web.ihep.su/library/pubs//prep1996/96-55-e.htm">IHEP Document Server</a>; <a  href="http://www.maik.rssi.ru/cgi-bin/search.pl?type=abstract&name=physpart&number=29/463">Phys.Part.Nucl. Server</a>


109056 <a href="https://lib-extopc.kek.jp/preprints/PDF/1919/1976/197607087.pdf"> KEK scanned document</a>; <a href="http://adsabs.harvard.edu/abs/1977PThPh..57..200I"> ADS Abstract Service</a>


402151 <a href="https://lib-extopc.kek.jp/preprints/PDF/1996/9607/9607052.pdf"> KEK scanned document</a>; <a href="http://adsabs.harvard.edu/abs/1996NuPhB.478....3H"> ADS Abstract Service</a>


218086 <a href="https://lib-extopc.kek.jp/preprints/PDF/1985/8509/8509256.pdf"> KEK scanned document</a>


318244 <a href="https://lib-extopc.kek.jp/preprints/PDF/1991/9109/9109060.pdf"> KEK scanned document</a>; <a  href="http://alice.cern.ch/format/showfull?sysnb=0137265">CERN Library Record</a>


214123 <a href="https://lib-extopc.kek.jp/preprints/PDF/1985/8507/8507206.pdf"> KEK scanned document</a>; <a href="http://adsabs.harvard.edu/abs/1985PhRvD..32.2901Y"> ADS Abstract Service</a>; <a  href="http://www-public.slac.stanford.edu/sciDoc/docMeta.aspx?slacPubNumber=SLAC-PUB-3628">SLAC Document Server</a>; <a  href="http://inspirehep.net/record/214123/files/slac-pub-3628.pdf">Link to Fulltext</a>


294909 <a href="https://lib-extopc.kek.jp/preprints/PDF/1919/1990/199005169.pdf"> KEK scanned document</a>; <a  href="http://alice.cern.ch/format/showfull?sysnb=0117760">CERN Library Record</a>

.....
```

File: bibformat/format_elements/bfe_INSPIRE_links.py (hunk 1 above)
Comment (kaplun) [inline bibformat/format_elements/bfe_INSPIRE_links.py]: Yep!

Comment (kaplun) [inline bibformat/format_elements/bfe_INSPIRE_links.py]: Woah Thanks for the thorugh testing!

**Comment by kaplun** [comment]
Yeah I discovered that not all the IDs corresponds to PDFs. Some of them are links that redirect to publisher(!)

**Comment by tsgit** [comment]
I'm looking at those that end up 1919

000301584 035__ $$9KEKSCAN$$a1990-12-240
gives
https://lib-extopc.kek.jp/preprints/PDF/1919/1990/199012240.pdf
but should be
https://lib-extopc.kek.jp/preprints/PDF/1990/9012/9012240.pdf

**Comment by kaplun** [comment]
Sigh. Have to make it smarter then. Looks like sometimes they might have used 4 digits sometimes 2, even before the year 2000.

**Comment by tsgit** [comment]
how about a post with content-type application/x-www-form-urlencoded with the identifier as form parameter IS_KEY_S1=200312019

like so
```
$ curl -d 'SUM_KIND=CsvSummary&SUM_NUMBER=50&IS_NUMBER=50&META_KIND=NOFRAME&IS_DB=G0000128Lib&IS_KIND=CsvInitSearch&IS_SCH=CSV&IS_TYPE=csv&DB_ID=G0000128Lib&GRP_ID=G0000128&IS_START=1&IS_TAG_S1=Record&SUM_TYPE=normal&IS_KEY_S1=200312019' https://www.i-repository.net/il/meta_pub/sresult
```

**Comment by kaplun** [comment]
That wouldn't lead to the PDF as we have right now, isn't it?

**Comment by tsgit** [comment]
Annette and Kirsten say the previous search link also didn't directly give a PDF but a splash page and that sometimes there is no PDF and instead there is a TIFF, but I have not found an example of that. What I do know is that there isn't always a PDF. 
I tried to reduce the hidden parameters here, I'm sure some more can be dropped
```
<html>
  <head/>
  <body>
    <form method="post" action="https://www.i-repository.net/il/meta_pub/sresult">
      <input type="text" name="IS_KEY_S1">
      <input type="submit" value="search">
      <input type="hidden" name="SUM_KIND" value="CsvSummary">
      <input type="hidden" name="SUM_NUMBER" value="50">
      <input type="hidden" name="IS_NUMBER" value="50">
      <input type="hidden" name="META_KIND" value="NOFRAME">
      <input type="hidden" name="IS_DB" value="G0000128Lib">
      <input type="hidden" name="IS_KIND" value="CsvInitSearch">
      <input type="hidden" name="IS_SCH" value="CSV">
      <input type="hidden" name="IS_TYPE" value="csv">
      <input type="hidden" name="DB_ID" value="G0000128Lib">
      <input type="hidden" name="GRP_ID" value="G0000128">
      <input type="hidden" name="IS_START" value="1">
      <input type="hidden" name="IS_TAG_S1" value="Record">
      <input type="hidden" name="SUM_TYPE" value="normal">
    </form>
  </body>
</html>
```

**Comment by tsgit** [comment]
anyhow, it's an option to consider. I'd prefer the direct PDF link if they are reliable

**Comment by tsgit** [comment]
the minimal number of hidden fields that works for me

```
<html>
  <head/>
  <body>
    <form method="post" action="https://www.i-repository.net/il/meta_pub/sresult">
      <input type="text" name="IS_KEY_S1">
      <input type="submit" value="search">
      <!--      <input type="hidden" name="META_KIND" value="NOFRAME">
		<input type="hidden" name="SUM_KIND" value="CsvSummary">
		<input type="hidden" name="SUM_NUMBER" value="50">
		<input type="hidden" name="SUM_TYPE" value="normal">
		-->

      <input type="hidden" name="IS_KIND" value="CsvInitSearch">
      <input type="hidden" name="IS_DB" value="G0000128Lib">  
      <input type="hidden" name="IS_TAG_S1" value="Record">
      <!--
	  <input type="hidden" name="IS_NUMBER" value="50">
	  <input type="hidden" name="IS_START" value="1">
	  -->
      <input type="hidden" name="IS_SCH" value="CSV">
      <input type="hidden" name="IS_TYPE" value="csv">

      <input type="hidden" name="DB_ID" value="G0000128Lib">
      <input type="hidden" name="GRP_ID" value="G0000128">
    </form>
  </body>
</html>
```

**Comment by tsgit** [comment]
so funnily, the PDF for the default kek-id I picked here http://inspirehep.net/info/hep/test/KEK-scan-search.html
is actually not available if you follow that link. :worried:

**Comment by tsgit** [comment]
@kaplun this https://github.com/tsgit/inspire/commit/ff7ee2defb88282950773a257b4f91c6262b4c5c addresses the issues I flagged with your PR. I checked all 66k ids, there are only 2 non-well-formed ones

3 digit year:
`000298148 035__ $$9KEKSCAN$$a200-32-881`

`000613180 035__ $$9KEKSCAN$$atest`

all others are either 7 or 9 digits. so the length check makes sense
and the startswith 19/20 is slightly redundant

I'm ready to deploy that if you agree

**Comment by kaplun** [comment]
Hi @tsgit thanks a lot for all this! Your fixes look great. :+1: for deploying them.

**Pull request closed without merging**